### ISSUE TITLE
docker images for all supported gentoo arches

### ISSUE THREAD

**Issue by paleozogt** [gentoo/gentoo-docker-images#75]
The `stage3.Dockerfile` pulls from `https://ftp-osl.osuosl.org/pub/gentoo/releases/`, which contains a lot more arches than just x86 and amd64.  With a few tweaks of the `build.sh` script, we could have docker images for any of the supported Gentoo arches (arm, ppc, et al).

**Comment by necrose99** [comment]
http://distfiles.gentoo.org/releases/arm/autobuilds/latest-stage3-arm64.txt
https://github.com/necrose99/Docker-Gentoo-ARM64 bit longish ago. 

multiarch/alpine:arm64-latest-stable 
gentoo and static Qemu also has been a pest .. for lowly admins like myself. 
**https://github.com/multiarch/qemu-user-static/releases** x86_64_qemu-aarch64-static.tar.gz
on local sure , likely but dockerhub not so much. 
I have lots of frustrating epic amouts of fails. 

building pkgs ie Rock64pro or rock64 laptop or rpi , then building a sysroot from scratch on device/s would be a plus. or detect native os ie avoid qt upgrade hell with bare stage3 .. mount packages to /packages use host/drop package on host.. 

**https://github.com/balena-io-library/armv7hf-debian-qemu**  useful wrapper to run on armhf qemu to amd64 etc..  could also be wrapped under alpine add go lang build push to /gentoo-arm64 builder temp , 
wappers /bin/bash or bin/sh with less awkwardness so users and other dockers built in dockerhub run cleanly and `transparently` .. 

 gentoo-arm64-catalyze'r host:/stages < docker drops stages to host..  shure devs would like these for faster autobuilds via docker.   or anyone else using previos docker images , migrate from to gentoo-arm64 etc. use rpi:3  for kiosk pc.. etc.. keeping nginx etc etc.. 

https://packages.gentoo.org/packages/app-emulation/qemu-riscv64-bin 
also easy to add .. for riscv64 or forking ebuild for other arches. ie arm64 
(https://github.com/multiarch/qemu-user-static/releases/download/v4.2.0-4/x86_64_qemu-aarch64-static.tar.gz ) dosym so quem wont collide if built

**Comment by KSmanis** [comment]
PR #92 adds CI builds for all stage3 architectures, except the following:
 * `amd64`
   * `stage3-amd64-hardened-selinux`: does SELinux make sense inside containers?
   * `stage3-amd64-hardened-selinux+nomultilib`: does SELinux make sense inside containers?
   * `stage3-x32`: not supported by Docker
 * `arm`
   * `stage3-armv4tl`: not supported by Docker
   * `stage3-armv5tel`: not officially supported by Docker
   * `stage3-armv6j`: stale stage3 tarball
   * `stage3-armv7a`: stale stage3 tarball
 * `ppc`
   * `stage3-ppc`: not supported by Docker
   * `stage3-ppc64`: not supported by Docker
 * `s390`
   * `stage3-s390`: not supported by Docker
 * `x86`
   * `stage3-i486`: not supported by Docker

Feel free to chime in if you feel like any of these targets can or should be supported as well.

**Comment by paleozogt** [comment]
While Docker can't be installed on some architectures (such as ppc), images for those arches can still be created and consumed by using qemu, which is what I've been doing.

**Comment by KSmanis** [comment]
Indeed, but is there any benefit in using QEMU emulation with a ppc image
rather than the ppc64le image which is supported?

**Comment by necrose99** [comment]
https://hub.docker.com/r/ppc64le/alpine/ @KSmanis


```
ARG BOOTSTRAP
FROM ${BOOTSTRAP:-alpine:3.11} as builder to ppc64le/alpine 
```
 could do or use various qemu enabled stage 3 gentoo builders.

`FROM ${BOOTSTRAP-${myarch}/:-alpine: as builder` 

as many fails as i've had for getting gentoo images to build on dockerhub ie arm64 ... i'd welcome a stable base image..

one irk for docker is  **Machine detection** ie IF machine = amd64 use emu
 **if native skip**... qemu (ie lucky enough to have an arm64 server or rpi cluster.. )  
(as an enthusiast or engineer getting devs notice can be hit or miss. )

**Comment by LinuxUserGD** [comment]
> PR #92 adds CI builds for all stage3 architectures, except the following:
> * `amd64`
> * `stage3-amd64-hardened-selinux`: [SELinux doesn't seem to make sense inside containers](https://serverfault.com/q/757606/)
> * `stage3-amd64-hardened-selinux+nomultilib`: [SELinux doesn't seem to make sense inside containers](https://serverfault.com/q/757606/)
> * `stage3-x32`: not supported by Docker

Didn't find a notice about llvm targets in the README, is it planned to support actions building llvm docker images?

amd64:
- [stage3-amd64-llvm-openrc](https://distfiles.gentoo.org/releases/amd64/autobuilds/current-stage3-amd64-llvm-openrc/)
- [stage3-amd64-llvm-systemd-mergedusr](https://distfiles.gentoo.org/releases/amd64/autobuilds/current-stage3-amd64-llvm-systemd-mergedusr/)
- [stage3-amd64-llvm-systemd](https://distfiles.gentoo.org/releases/amd64/autobuilds/current-stage3-amd64-llvm-systemd/)
- [stage3-amd64-musl-llvm](https://distfiles.gentoo.org/releases/amd64/autobuilds/current-stage3-amd64-musl-llvm/)

arm64:
- [stage3-arm64-llvm-openrc](https://distfiles.gentoo.org/releases/arm64/autobuilds/current-stage3-arm64-llvm-openrc/)
- [stage3-arm64-llvm-systemd](https://distfiles.gentoo.org/releases/arm64/autobuilds/current-stage3-arm64-llvm-systemd/)
- [stage3-arm64-llvm-systemd-mergedusr](https://distfiles.gentoo.org/releases/arm64/autobuilds/current-stage3-arm64-llvm-systemd-mergedusr/)
- [stage3-arm64-musl-llvm](https://distfiles.gentoo.org/releases/arm64/autobuilds/current-stage3-arm64-musl-llvm/)
